petition_service/internal/controller/rpc/petition_controller.go

```diff
@@ -228,3 +232,73 @@ func (s *Server) GetUserVotedPetitions(_ context.Context, req *pb.GetUserVotedPe
 		Petitions: getUserPetitionsResponse,
 	}, nil
 }
+
+func (s *Server) CheckIfPetitionsExpired(_ context.Context, req *pb.Petition) (*empty.Empty, error) {
+	petition, err := s.PetitionService.GetByID(uint(req.Id))
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			slog.Errorf("Petition %v not found", req.Id)
+			return nil, status.Error(codes.NotFound, "petition not found")
+		}
+		return nil, err
+	}
+	slog.Info("Petition %v successfully found", req.Id)
+
+	if _, err := s.PetitionService.CheckPetitionExpiration(petition); err != nil {
+		return nil, err
+	}
+
+	slog.Info("Petition %v expiration date successfully checked", req.Id)
+	return &empty.Empty{}, nil
+}
+
+func ScheduleDailyCheck(s *Server) {
```

Comment:
Imo the ScheduleDailyCheck would fit better at the service layer. The gRPC  layer does not need to know the logic about how and when we check the petitions's expiration date.